src/plugins/vis_types/vega/public/vega_view/vega_base_view.js

```diff
@@ -57,8 +57,25 @@ function normalizeDate(date) {
   return normalizedDate;
 }
 
+// recursive function to check a nested object for the 'toJSON' property
+function checkObjectForToJsonProperty(object) {
+  let objHasToJsonProperty = false;
+  Object.keys(object).forEach((key) => {
+    if (typeof object[key] === 'object') {
```

Comment:
Doesn't check for `null` which causes an error one level deeper because `Object.keys(null)` throws